codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-batchmanager-sync-class.java

```diff
@@ -0,0 +1,97 @@
+package software.amazon.awssdk.services.batchmanager;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import software.amazon.awssdk.annotations.Generated;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.RequestOverrideConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientOption;
+import software.amazon.awssdk.core.client.handler.SyncClientHandler;
+import software.amazon.awssdk.core.http.HttpResponseHandler;
+import software.amazon.awssdk.metrics.MetricPublisher;
+import software.amazon.awssdk.protocols.json.AwsJsonProtocol;
+import software.amazon.awssdk.protocols.json.AwsJsonProtocolFactory;
+import software.amazon.awssdk.protocols.json.BaseAwsJsonProtocolFactory;
+import software.amazon.awssdk.protocols.json.JsonOperationMetadata;
+import software.amazon.awssdk.services.batchmanager.model.BatchManagerException;
+import software.amazon.awssdk.services.batchmanagertest.batchmanager.SyncBatchManagerTest;
+import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.ThreadFactoryBuilder;
+
+/**
+ * Internal implementation of {@link BatchManagerClient}.
+ *
+ * @see BatchManagerClient#builder()
+ */
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+final class DefaultBatchManagerClient implements BatchManagerClient {
+    private static final Logger log = Logger.loggerFor(DefaultBatchManagerClient.class);
+
+    private final SyncClientHandler clientHandler;
+
+    private final AwsJsonProtocolFactory protocolFactory;
+
+    private final SdkClientConfiguration clientConfiguration;
+
+    private final ScheduledExecutorService executorService;
+
+    private final ExecutorService executor;
+
+    protected DefaultBatchManagerClient(SdkClientConfiguration clientConfiguration) {
+        this.clientHandler = new AwsSyncClientHandler(clientConfiguration);
+        this.clientConfiguration = clientConfiguration;
+        this.protocolFactory = init(AwsJsonProtocolFactory.builder()).build();
+        this.executorService = clientConfiguration.option(SdkClientOption.SCHEDULED_EXECUTOR_SERVICE);
+        ThreadFactory threadFactory = new ThreadFactoryBuilder().threadNamePrefix("DefaultBatchManagerClient").build();
+        this.executor = Executors.newSingleThreadExecutor(threadFactory);
```

Comment:
We should only create it in the SqsBatchManager class and not the client class. Also, single thread executor might not be the best choice here since all requests will be blocking call.  Let's create the same one as defined here https://github.com/aws/aws-sdk-java-v2/blob/39d6f2a7d3c98bffff57c574c5bf1c63bc72e698/services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/S3NativeClientConfiguration.java#L118

Reply:
Ok that makes sense, I'll remove it from the codegen and add it in the handcoded SqsBatchManager for now.